api/data_explorer/util/elasticsearch_util.py

```diff
@@ -215,13 +251,44 @@ def get_elasticsearch_facet(es, elasticsearch_field_name, field_type):
             field=elasticsearch_field_name,
             interval=_get_bucket_interval(field_range))
 
-    # Handle sample facets in a special way since they are nested objects.
-    if elasticsearch_field_name.startswith('samples.'):
-        es_facet = NestedFacet('samples', es_facet)
-
+    nested_facet = _get_nested_facet(elasticsearch_field_name, es_facet,
+                                     nested_paths)
+    if nested_facet:
+        es_facet = nested_facet
     return es_facet
 
 
+def _get_nested_facet(elasticsearch_field_name, es_facet, nested_paths):
+    """
+    Returns a NestedFacet for the Elasticsearch field, if the field is nested.
+    """
+    parent = elasticsearch_field_name.rsplit('.', 1)[0]
+    nested_facet = None
+    # Traverse up the nesting levels from the leaf field, till we reach the root.
+    # Need to traverse till the root, because the root can be a nested field,
+    # for example "samples". All the sub fields can be non-nested, like
+    # "samples.verily-public-data.human_genome_variants.1000_genomes_sample_info.Main_project_LC_platform"
+    # This field needs to be a NestedFacet because an ancestor("samples") is nested.
+    while parent:
```

Comment:
Here's a simpler version, does this work?
```
def _get_nested_facet(elasticsearch_field_name, es_facet, nested_paths):
    """
    Returns a NestedFacet for the Elasticsearch field, if the field is nested.
    """
    parts = elasticsearch_field_name.rsplit('.', 1)
    # Traverse up the nesting levels from the leaf field, till we reach the root.
    # Need to traverse till the root, because the root can be a nested field,
    # for example "samples". All the sub fields can be non-nested, like
    # "samples.verily-public-data.human_genome_variants.1000_genomes_sample_info.Main_project_LC_platform"
    # This field needs to be a NestedFacet because an ancestor("samples") is nested.
    while len(parts) > 1:
        parent = parts[0]
        if parent in nested_paths:
            es_facet = NestedFacet(parent, es_facet)
        parts = parent.rsplit('.', 1)

    return es_facet
```

Reply:
Yeah that is simpler, Python simplifies array search.